web/src/components/tasks/business-formation/business/NonprofitProvisions.test.tsx

```diff
@@ -0,0 +1,152 @@
+      expect(screen.getByLabelText("Nonprofit board member rights terms")).toBeInTheDocument();
+    });
+
+    it("does not show text field when IN_BYLAWS is selected", async () => {
+      const page = await getPageHelper(
+        { legalStructureId: "nonprofit" },
+        { hasNonprofitBoardMembers: true, nonprofitBoardMemberRightsSpecified: undefined }
+      );
+      expect(screen.queryByLabelText("Nonprofit board member rights terms")).not.toBeInTheDocument();
+      page.chooseRadio("nonprofitBoardMemberRightsSpecified-IN_BYLAWS");
+      expect(screen.queryByLabelText("Nonprofit board member rights terms")).not.toBeInTheDocument();
+    });
+  });
+
+  describe("nonprofitTrusteesMethodSpecified", () => {
+    it("shows text field when IN_FORM is selected", async () => {
+      const page = await getPageHelper(
+        { legalStructureId: "nonprofit" },
+        { hasNonprofitBoardMembers: true, nonprofitTrusteesMethodSpecified: undefined }
+      );
+      expect(screen.queryByLabelText("Nonprofit trustees method terms")).not.toBeInTheDocument();
+      page.chooseRadio("nonprofitTrusteesMethodSpecified-IN_FORM");
+      expect(screen.getByLabelText("Nonprofit trustees method terms")).toBeInTheDocument();
+    });
+
+    it("does not show text field when IN_BYLAWS is selected", async () => {
+      const page = await getPageHelper(
+        { legalStructureId: "nonprofit" },
+        { hasNonprofitBoardMembers: true, nonprofitTrusteesMethodSpecified: undefined }
+      );
+      expect(screen.queryByLabelText("Nonprofit trustees method terms")).not.toBeInTheDocument();
+      page.chooseRadio("nonprofitTrusteesMethodSpecified-IN_BYLAWS");
+      expect(screen.queryByLabelText("Nonprofit trustees method terms")).not.toBeInTheDocument();
+    });
+  });
+
+  describe("nonprofitAssetDistributionSpecified", () => {
+    it("shows text field when IN_FORM is selected", async () => {
+      const page = await getPageHelper(
+        { legalStructureId: "nonprofit" },
+        { hasNonprofitBoardMembers: true, nonprofitAssetDistributionSpecified: undefined }
+      );
+      expect(screen.queryByLabelText("Nonprofit asset distribution terms")).not.toBeInTheDocument();
+      page.chooseRadio("nonprofitAssetDistributionSpecified-IN_FORM");
+      expect(screen.getByLabelText("Nonprofit asset distribution terms")).toBeInTheDocument();
+    });
+
+    it("does not show text field when IN_BYLAWS is selected", async () => {
+      const page = await getPageHelper(
+        { legalStructureId: "nonprofit" },
+        { hasNonprofitBoardMembers: true, nonprofitAssetDistributionSpecified: undefined }
+      );
+      expect(screen.queryByLabelText("Nonprofit asset distribution terms")).not.toBeInTheDocument();
+      page.chooseRadio("nonprofitAssetDistributionSpecified-IN_BYLAWS");
+      expect(screen.queryByLabelText("Nonprofit asset distribution terms")).not.toBeInTheDocument();
+    });
+  });
+});
```

Comment:
[boulder] This whole test appears to be missing a describe block describing the behavior of `hasNonprofitBoardMembers`, if I'm not missing something? There's a missing test for what happens if that field is true, false, and not yet specified, which is huge because it controls the behavior of all the other fields on the component.

Reply:
Done